Source/PlayField.cpp

```diff
@@ -0,0 +1,151 @@
+void PlayField::print() const {
+    for(int i = 0; i < HEIGHT_FIELD; ++i) {
+        for(int j = 0; j < HEIGHT_FIELD; ++j) {
+            char symbol;
+            if(cellField[CellPos(i, j)] == csCross)
+                symbol = 'X';
+            else if(cellField[CellPos(i, j)] == csNought)
+                symbol = '0';
+            else
+                symbol = ' ';
+            std::cout<<"|"<<symbol;
+        }
+        std::cout<<"|"<<std::endl;
+    }
+}
+
+PlayField PlayField::operator+(PlayField::CellPos item)  const {
+    auto countCross = 0;
+    auto countNought = 0;
+
+    for(auto elem : cellField)
+        if(elem == csCross)
+            countCross++;
+        else if(elem == csNought)
+            countNought++;
+
+    auto result = *this;
+    result.cellField[item] = (countCross - countNought == 1) ? csNought : csCross;
+    return result;
+}
+
+std::vector<PlayField::CellPos> PlayField::getEmptyCells() const {
+    std::vector<CellPos> result;
+    for (int i = 0; i < SIZE_FIELD ; ++i)
+        if(cellField[i] == csEmpty)
+            result.emplace_back(i / HEIGHT_FIELD, i % HEIGHT_FIELD);
+    return result;
+}
+
+PlayField PlayField::makeMove(PlayField::CellPos item) const {
+    assert(cellField[item] == csEmpty && checkFieldStatus() == fsNormal);
+    return  *this + item;
+}
+
+PlayField::fnState PlayField::checkFieldStatus() const {
+    auto countCross = 0;
+    auto countNought = 0;
+    auto winCross = false;
+    auto winNought = false;
+
+    for(auto elem : cellField)
+        if(elem == csCross)
+            countCross++;
+        else if(elem == csNought)
+            countNought++;
+
+    auto diff = countCross-countNought;
+    if(0 > diff || diff > 1)
+        return fsInvalid;
+
+    // Проверяем горизонталь
+    for(int i = 0; i < HEIGHT_FIELD; ++i) {
+        bool horizontal = true;
+        for(int j = 1; j < HEIGHT_FIELD; ++j) {
+            auto flag = cellField[CellPos(i, j - 1)] == cellField[CellPos(i, j)];
+            horizontal = (horizontal) ? flag : false;
+        }
+        if(horizontal && cellField[CellPos(i, 0)] == csNought)
+            winNought = true;
+        if(horizontal && cellField[CellPos(i, 0)] == csCross)
+            winCross = true;
+    }
+    // Проверяем вертикаль
+    for(int i = 0; i < HEIGHT_FIELD; ++i) {
+        bool vertical = true;
+        for(int j = 1; j < HEIGHT_FIELD; ++j) {
+            auto flag = cellField[CellPos(j - 1, i)] == cellField[CellPos(j, i)] ;
+            vertical = (vertical) ? flag : false;
+        }
+        if(vertical && cellField[CellPos(0, i)] == csNought)
+            winNought = true;
+        if(vertical && cellField[CellPos(0, i)] == csCross)
+            winCross = true;
+    }
+    // Главная диагональ
+    int i = 1;
+    int j = 1;
+    bool mainDiagonal = true;
+    while(i < HEIGHT_FIELD){
+        auto flag = cellField[CellPos(i - 1, j - 1)] == cellField[CellPos(i, j)];
+        mainDiagonal = (mainDiagonal) ? flag : false;
```

Comment:
Мне эта конструкция не нравится ни здесь, ни выше, ни ниже по коду.
```suggestion
        if(cellField[CellPos(i - 1, i - 1)] != cellField[CellPos(i, i)])
            mainDiagonal = false;
```

Reply:
Могу вообще так  `mainDiagonal = (mainDiagonal) ? cellField[CellPos(i - 1, j - 1)] == cellField[CellPos(i, j)] : false;` в одну строку

Reply:
Очень просто и понятно

Reply:
Если честно -- ещё хуже. Мало того, что тернарный оператор записан в одну строку, дак ещё и условие стало большим. + опять же вычурная логика. А по быстродействию ни чуть не лучше. А в моём случае добавить ещё break под условие -- и будет вообще красота.